pages/myposts.vue

```diff
@@ -244,314 +255,250 @@
     </b-container>
   </client-only>
 </template>
-<script>
+
+<script setup>
 import { useRoute } from 'vue-router'
 import pluralize from 'pluralize'
 import dayjs from 'dayjs'
-import { useGroupStore } from '../stores/group'
 import { useAuthStore } from '../stores/auth'
 import { useMessageStore } from '../stores/message'
 import { useMiscStore } from '../stores/misc'
-import { useComposeStore } from '../stores/compose'
 import { useSearchStore } from '../stores/search'
-import VisibleWhen from '~/components/VisibleWhen'
 import { buildHead } from '~/composables/useBuildHead'
+
+import VisibleWhen from '~/components/VisibleWhen'
 import InfiniteLoading from '~/components/InfiniteLoading'
-const JobsTopBar = () => import('~/components/JobsTopBar')
-const MyMessage = () => import('~/components/MyMessage.vue')
-const SidebarLeft = () => import('~/components/SidebarLeft')
-const SidebarRight = () => import('~/components/SidebarRight')
-const DonationAskModal = () => import('~/components/DonationAskModal')
-const ExpectedRepliesWarning = () =>
-  import('~/components/ExpectedRepliesWarning')
+import SidebarLeft from '~/components/SidebarLeft'
+import SidebarRight from '~/components/SidebarRight'
+import ExpectedRepliesWarning from '~/components/ExpectedRepliesWarning'
+import JobsTopBar from '~/components/JobsTopBar'
+import MyMessage from '~/components/MyMessage.vue'
+import UserSearch from '~/components/UserSearch.vue'
+import DonationAskModal from '~/components/DonationAskModal'
+
+const authStore = useAuthStore()
+const messageStore = useMessageStore()
+const searchStore = useSearchStore()
+const miscStore = useMiscStore()
+
+const runtimeConfig = useRuntimeConfig()
+const route = useRoute()
 
 definePageMeta({
   layout: 'login',
 })
 
-export default {
-  components: {
-    VisibleWhen,
-    JobsTopBar,
-    MyMessage,
-    SidebarLeft,
-    SidebarRight,
-    DonationAskModal,
-    ExpectedRepliesWarning,
-    InfiniteLoading,
-  },
-  mixins: [buildHead],
-  async setup() {
-    const runtimeConfig = useRuntimeConfig()
-    const route = useRoute()
-
-    useHead(
-      buildHead(
-        route,
-        runtimeConfig,
-        'My Posts',
-        "See OFFERs/WANTEDs that you've posted, and replies to them.",
-        null,
-        {
-          class: 'overflow-y-scroll',
-        }
-      )
-    )
-
-    const authStore = useAuthStore()
-    const messageStore = useMessageStore()
-    const groupStore = useGroupStore()
-    const miscStore = useMiscStore()
-    const composeStore = useComposeStore()
-    const searchStore = useSearchStore()
-
-    // We want this to be our next home page.
-    const existingHomepage = miscStore.get('lasthomepage')
-
-    if (existingHomepage !== 'myposts') {
```

Comment:
This block is removed:

1. I don't understand what it's for
2. If it's needed for some later time, let's re-add it at that later time. No need to pullute the existing code base.

Reply:
Some people like the home page (the page we load when they visit the site) to be My Posts, some like it to be Browse, some people like it to be ChitChat.

So we store this persistently on the client - see `misc.js` for the persisting strategy.  Each of these potential home pages updates that value.  Then in `pages/index.vue` we read it and then route / to the appropriate page, which means you are taken to the last of these pages which you visited.

So please add this back in.

Reply:
Thanks for the explanations! Reverted.